include/rdma/fi_var.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2015, Cisco Systems, Inc. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#ifndef _FI_VAR_H_
+#define _FI_VAR_H_
+
+#include <rdma/fi_prov.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Registers a configuration variable for use with libfabric.
+ *
+ * Example: fi_register_var(provider, "foo", "Very important help
+ * string");
+ *
+ * This registers the configuration variable "foo" in the specified
+ * provider.
+ *
+ * The help string cannot be NULL or empty.
+ *
+ * The var_name and help_string parameters will be copied internally;
+ * they can be freed upon return from fi_var_register().
+ */
+int fi_var_register(const struct fi_provider *provider, const char *var_name,
+		const char *help_string);
+
+/* Get the value of an configuration variable.
+ *
+ * Currently, configuration variables will only be read from the
+ * environment.  The environment variable names will be of the form
+ * upper_case(FI_<provider_name>_<var_name>).
+ *
+ * Someday this call could be expanded to also check config files.
+ *
+ * Upon successful return, FI_SUCCESS is returned and (*value) points
+ * to a \0-terminated string value, or NULL if it was not set.  The
+ * caller should not modify or free (*value).
+ *
+ * FI_ENOENT will be returned if var_name was not previously
+ * registered via fi_var_register().
+ */
+int fi_var_get(struct fi_provider *provider, const char *var_name,
+	char **value);
```

Comment:
Mmm.  Good point.  I don't have an opinion here -- I'm happy to add the following:

* `_get_str()`
* `_get_int()`
* `_get_long()` 
* `_get_bool()` (which takes 0, true, yes, on, 1, false, no, off -- case insensitive, of course)

That would cover all the cases in sockets and psm.

Want me to add these?